src/Models/Model.php

```diff
@@ -57,6 +61,35 @@ public function __construct( array $attributes = [] ) {
 		$this->syncOriginal();
 	}
 
+	/**
+	 * Casts the value for the type, used when constructing a model from query data. If the model needs to support
+	 * additional types, especially class types, this method can be overridden.
+	 *
+	 * @since 2.0.0 changed to static
+	 *
+	 * @param string $type
+	 * @param mixed  $value The query data value to cast, probably a string.
+	 * @param string $property The property being casted.
+	 *
+	 * @return mixed
+	 */
+	protected static function castValueForProperty( string $type, $value, string $property ) {
+		switch ( $type ) {
+			case 'int':
+				return (int) $value;
+			case 'string':
+				return (string) $value;
+			case 'bool':
+				return (bool) $value;
+			case 'array':
+				return (array) $value;
+			case 'float':
+				return (float) $value;
+			default:
+				Config::throwInvalidArgumentException( "Unexpected type: '$type'. To support additional types, implement a custom castValueForProperty() method." );
```

Comment:
I feel like this could happen up the chain before we get into casting, doing something like `$this->isPropertyTypeValid($key, $value)`

Reply:
I purposely want to do this here so all someone has to do is overload this function to support additional casting of types.